### PR TITLE
Project edit issue

### PR DESCRIPTION
**Pull request by abinashpa**
## Notion card
https://www.notion.so/saeloun/6afc9500b2cc42af86be55f37894fb09?v=f1810139870444259ef38b4bee169b34&p=2e3337478aec42818dbaca5f4ba3c1f5

## Summary
The page is breaking after clicking on edit project details

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx [hunk 1]
```diff
@@ -8,44 +9,43 @@ const AddEditProject = ({
...
-    if (editProjectData) {
-      if (clientList) {
-        const client = clientList.filter(clientItem => clientItem.name == editProjectData.clientName);
-        setClient(client[0].id);
-      }
-      setProjectName(editProjectData ? editProjectData.name : null);
-      setProjectType(editProjectData.isBillable ? "Billable" : "Non-Billable");
+    if (! editProjectData) return;
+    if (clientList.length) {
+      const client = clientList.filter(clientItem => clientItem["name"] == editProjectData["client"]["name"]);
```
Comment (ajinkyaa) [inline app/javascript/src/components/Projects/Modals/AddEditProject.tsx:44]: why not use `search`

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx [hunk 2]
```diff
@@ -106,7 +106,7 @@ const AddEditProject = ({
                   <select
                     defaultValue={client}
                     className="rounded border-0 block w-full px-2 py-1 bg-miru-gray-100 h-8 font-medium text-sm text-miru-dark-purple-1000 focus:outline-none sm:text-base"
-                    onChange={(e) => setClient(e.target.value)}>
+                    onChange={(e) => setClient(+ e.target.value)}>
```
Comment (akhilgkrishnan) [inline app/javascript/src/components/Projects/Modals/AddEditProject.tsx:109]: What's the use of + here?

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx [hunk 3]
```diff
@@ -8,44 +9,43 @@ const AddEditProject = ({
...
 
   useEffect(() => {
-    if (editProjectData) {
-      if (clientList) {
-        const client = clientList.filter(clientItem => clientItem.name == editProjectData.clientName);
-        setClient(client[0].id);
-      }
-      setProjectName(editProjectData ? editProjectData.name : null);
-      setProjectType(editProjectData.isBillable ? "Billable" : "Non-Billable");
+    if (! editProjectData) return;
```
[the remark this answers is not in the document]
Comment (abinashpa) [reply app/javascript/src/components/Projects/Modals/AddEditProject.tsx:42]: No, It's just more elegant than nested if condition.

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx (hunk 2 above)
Comment (abinashpa) [reply app/javascript/src/components/Projects/Modals/AddEditProject.tsx:109]: Converting string to a number.

**Comment by supriya3105** [comment]
@ajinkyaa @akhilgkrishnan Please review this PR. Abinash has responded to your questions. 
@apoorv1316 Please test this.

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx (hunk 1 above)
Comment (abinashpa) [reply app/javascript/src/components/Projects/Modals/AddEditProject.tsx:44]: Do you mean `Array.find`?

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@abinashpa Screenshot 2022-06-17 at 9 29 14 AM
 Getting this error on the projects page. Please fix it.

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx [hunk 4]
```diff
@@ -106,7 +107,7 @@ const AddEditProject = ({
                   <select
                     defaultValue={client}
                     className="rounded border-0 block w-full px-2 py-1 bg-miru-gray-100 h-8 font-medium text-sm text-miru-dark-purple-1000 focus:outline-none sm:text-base"
-                    onChange={(e) => setClient(e.target.value)}>
+                    onChange={(e) => setClient(+ e.target.value)}>
                     <option value='0'>Select Client</option>
                     {clientList &&
                       clientList.map((e, index) => <option key={index} value={e.id} selected={e.id == client}>{e.name}</option>)}
```
Comment (akhilgkrishnan) [inline app/javascript/src/components/Projects/Modals/AddEditProject.tsx:113]: user proper naming instead of e

**Review by akhilgkrishnan — COMMENTED** [review]
@abinashpa what's the use of helix file?

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx (hunk 4 above)
Comment (abinashpa) [reply app/javascript/src/components/Projects/Modals/AddEditProject.tsx]: Changed to event.

File: app/javascript/src/components/Projects/Modals/AddEditProject.tsx [hunk 5]
```diff
@@ -8,44 +9,44 @@ const AddEditProject = ({
...
   useEffect(() => {
-    if (editProjectData) {
-      if (clientList) {
-        const client = clientList.filter(clientItem => clientItem.name == editProjectData.clientName);
-        setClient(client[0].id);
-      }
-      setProjectName(editProjectData ? editProjectData.name : null);
-      setProjectType(editProjectData.isBillable ? "Billable" : "Non-Billable");
+    if (! editProjectData) return;
+    if (clientList.length) {
```
Comment (ajinkyaa) [inline app/javascript/src/components/Projects/Modals/AddEditProject.tsx:44]: what if clientList.length is `0`. What will happen?

**Review by ajinkyaa — APPROVED** [review]

**Comment by supriya3105** [comment]
@apoorv1316 Please test this and merge if the functionality is working.

**Comment by apoorv1316** [comment]
@supriya3105 2-3 things need to be fixed. I have talked and conveyed it to Abhinash already and he is working on it.

**Comment by supriya3105** [comment]
> @supriya3105 2-3 things need to be fixed. I have talked and conveyed it to Abhinash already and he is working on it.

Please add the feedback your have provided to the PR.

**Review by akhilgkrishnan — APPROVED** [review]
LGTM

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**